### ISSUE TITLE
[Help] How to use the IN operator with prepared statements?

### ISSUE THREAD

**Issue by rafaell-lycan** [brianc/node-postgres#2853]
Hello 👋 

I'm having a doubt on the best way to use the following query via the [prepared statements](https://www.postgresql.org/docs/9.3/static/sql-prepare.html):

```sql
SELECT *
  FROM table
WHERE ID IN (id1, id2, ...) 
```

What I'm trying is:
```js
const ids = ['01GG7E75ZBMSQQ6FPG5456ZHRP', '01GG7EY1XRJKF0KTA4GVZMST6A', '01GG7XHTVC5003RAVBENMC96CP']
const result = await this.db.query({
      name: 'get-by-ids',
      text: `SELECT * FROM table WHERE ID IN ($1)`,
      values: [ids],
})
```

But doesn't return anything or an error. Any thoughts?

**Comment by abdulloooh** [comment]
sorry to come back to this but what about `NOT IN` operation

**Comment by brianc** [comment]
`WHERE ID != ANY($1)`? Just guessing here...googling this answer is probably your best bet as this is just a postgres query syntax thing not really related to the node-postgres code

**Comment by abdulloooh** [comment]
alright thanks @brianc 
I tried that but didn't work

I eventually use parameterised query like you cited as alternative in the FAQ

**Comment by sehrope** [comment]
For the NOT IN case you need to use: `id != ALL($1)` ("id is not equal to all of these values")

SQL with `id != ANY($1)` would not work as if there's more than one value in $1, then id will definitely not be equal to one of those values.

For example if $1 = [1,2,3, 4], then regardless of the value of "id", it will always be != to at least one of those values as even if it matches one value, it definitely does not match the rest.

So in summary:

* If it's `=`, then use `ANY(...)`
* If it's `!=`, then use `ALL(...)`

**Comment by Zhilcov** [comment]
hello, is anybody know how to deal with multi key id 
`SELECT *
            FROM table
            WHERE (id1, id2) IN (('value1', 'value2'), ('value11', 'value22'))`
 to use ANY and prepared statements
